### PR TITLE
fix: Add Service Worker to the dev-bundle and serve it

### PR DESCRIPTION
**Pull request by mshabarov**
Related-to https://github.com/vaadin/flow/issues/15970

### CODE REVIEW

**Review by Artur- — DISMISSED** [review]

File: flow-server/src/main/java/com/vaadin/flow/server/StaticFileServer.java [hunk 1]
```diff
@@ -261,13 +261,14 @@ public boolean serveStaticResource(HttpServletRequest request,
         URL resourceUrl = null;
         if (!deploymentConfiguration.isProductionMode()
                 && !deploymentConfiguration.frontendHotdeploy()) {
-            if (filenameWithPath.startsWith(EXPRESS_MODE_BUNDLE_PATH_PREFIX)) {
+            if (filenameWithPath.startsWith(EXPRESS_MODE_BUNDLE_PATH_PREFIX)
```
Comment (Artur-) [inline flow-server/src/main/java/com/vaadin/flow/server/StaticFileServer.java:264]: This now matches /VAADIN/dev-bundle-abc/foo.js. Probably should not?

Comment (mshabarov) [reply flow-server/src/main/java/com/vaadin/flow/server/StaticFileServer.java]: Changed to match /VAADIN/dev-bundle/, thanks!

File: flow-server/src/main/java/com/vaadin/flow/server/StaticFileServer.java [hunk 2]
```diff
@@ -261,13 +263,14 @@ public boolean serveStaticResource(HttpServletRequest request,
         URL resourceUrl = null;
         if (!deploymentConfiguration.isProductionMode()
                 && !deploymentConfiguration.frontendHotdeploy()) {
-            if (filenameWithPath.startsWith(EXPRESS_MODE_BUNDLE_PATH_PREFIX)) {
+            if (filenameWithPath.startsWith(EXPRESS_MODE_BUNDLE_PATH)
+                    || filenameWithPath.startsWith("/sw.js")) {
                 // Express mode bundle file
                 String filenameInsideBundle = filenameWithPath
-                        .substring(EXPRESS_MODE_BUNDLE_PATH_PREFIX.length());
+                        .replaceFirst(EXPRESS_MODE_BUNDLE_PATH_PREFIX, "");
```
Comment (Artur-) [inline flow-server/src/main/java/com/vaadin/flow/server/StaticFileServer.java:270]:
```suggestion
                        .replaceFirst(EXPRESS_MODE_BUNDLE_PATH, "/");
```

Comment (Artur-) [reply flow-server/src/main/java/com/vaadin/flow/server/StaticFileServer.java:270]: Does not look like you need two separate constants

**Review by Artur- — CHANGES_REQUESTED** [review]

File: flow-server/src/main/java/com/vaadin/flow/server/StaticFileServer.java (hunk 2 above)
Comment (mshabarov) [reply flow-server/src/main/java/com/vaadin/flow/server/StaticFileServer.java]: Nice suggestion, thanks! Done

**Review by Artur- — APPROVED** [review]

**Pull request merged**